### PR TITLE
feat(vehicle_gate_pass): Created unpaid sales invoice checkbox

### LINKED ISSUE #345: feat(Email): System email for the specific users if vehicle realise on credit
When a vehicle is released after service on credit (billed but not paid), automatic emails are sent to the assigned user.

Email will sent to the users on the creation of the gate pass.

### PR DESCRIPTION
**Pull request by MAunZaidi**
**Feat**

"Closes #345"

### CODE REVIEW

File: erpnext/vehicles/doctype/vehicle_gate_pass/vehicle_gate_pass.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
   "sales_person",
   "service_advisor",
   "sales_invoice",
+  "invoice_unpaid_check",
```
Comment (SaiFi0102) [inline erpnext/vehicles/doctype/vehicle_gate_pass/vehicle_gate_pass.json]: Rename field to invoice_is_unpaid ... Refactor everywhere

File: erpnext/vehicles/doctype/vehicle_gate_pass/vehicle_gate_pass.py [hunk 2]
```diff
@@ -287,9 +289,20 @@ def add_vehicle_maintenance_schedule(self):
...
+		if self.get("sales_invoice"):
+			sales_invoice = frappe.db.get_value("Sales Invoice", self.sales_invoice, ['outstanding_amount', 'docstatus'],
+				as_dict=1)
+			if sales_invoice.outstanding_amount > 0 and sales_invoice.docstatus == 1:
+				self.invoice_unpaid_check = 1
+			else:
+				self.invoice_unpaid_check = 0
 
-@frappe.whitelist()
+@frappe.whitelist()
```
Comment (SaiFi0102) [inline erpnext/vehicles/doctype/vehicle_gate_pass/vehicle_gate_pass.py]: remove unnecessary change

File: erpnext/vehicles/doctype/vehicle_gate_pass/vehicle_gate_pass.py [hunk 3]
```diff
@@ -9,6 +9,7 @@
 from erpnext.maintenance.doctype.maintenance_schedule.maintenance_schedule import schedule_next_project_template
 
 
+
```
Comment (SaiFi0102) [inline erpnext/vehicles/doctype/vehicle_gate_pass/vehicle_gate_pass.py]: remove unnecessary change

File: erpnext/vehicles/doctype/vehicle_gate_pass/vehicle_gate_pass.py [hunk 4]
```diff
@@ -23,6 +24,7 @@ def validate(self):
 		self.validate_opportunity()
 		self.validate_vehicle_received()
 		self.validate_sales_invoice()
+		self.validate_invoice_unpaid_check()
```
Comment (SaiFi0102) [inline erpnext/vehicles/doctype/vehicle_gate_pass/vehicle_gate_pass.py]: rename to `set_invoice_is_unpaid()`

**Review by SaiFi0102 — CHANGES_REQUESTED** [review]

File: erpnext/vehicles/doctype/vehicle_gate_pass/vehicle_gate_pass.json [hunk 5]
```diff
@@ -427,13 +428,20 @@
    "label": "Customer Contact Name",
    "no_copy": 1,
    "read_only": 1
+  },
+  {
+   "default": "0",
+   "fieldname": "invoice_unpaid_check",
```
Comment (SaiFi0102) [inline erpnext/vehicles/doctype/vehicle_gate_pass/vehicle_gate_pass.json]: Still not renamed

**Review by SaiFi0102 — CHANGES_REQUESTED** [review]

File: erpnext/vehicles/doctype/vehicle_gate_pass/vehicle_gate_pass.json (hunk 5 above)
Comment (MAunZaidi) [reply erpnext/vehicles/doctype/vehicle_gate_pass/vehicle_gate_pass.json]: Done

File: erpnext/vehicles/doctype/vehicle_gate_pass/vehicle_gate_pass.py (hunk 3 above)
Comment (SaiFi0102) [inline erpnext/vehicles/doctype/vehicle_gate_pass/vehicle_gate_pass.py]: Remove unnecessary new line

**Review by SaiFi0102 — CHANGES_REQUESTED** [review]

**Review by SaiFi0102 — APPROVED** [review]

**Pull request merged**